### PR TITLE
FIX #6980 FormMail::getEmailTemplate returns an array instead of a ModelMail object

### LINKED ISSUE #6980: Bug: FormMail::getEmailTemplate returns an array instead of a ModelMail object

### PR DESCRIPTION
**Pull request by marcosgdf**
Close #6980

### CODE REVIEW

File: htdocs/core/class/html.formmail.class.php [hunk 1]
```diff
@@ -307,7 +305,7 @@ function get_form($addfileaction='addfile',$removefileaction='removefile')
 			if (GETPOST('mode','alpha') == 'init' || (GETPOST('modelmailselected','alpha') && GETPOST('modelmailselected','alpha') != '-1'))
 			{
 				$this->clear_attached_files();
-				if (! empty($arraydefaultmessage['joinfiles']) && is_array($this->param['fileinit']))
+				if (! empty($arraydefaultmessage->joinfiles) && is_array($this->param['fileinit']))
```
Comment (eldy) [inline htdocs/core/class/html.formmail.class.php]: $arraydefaultmessage is an array, why removing ['joinfiles'] ?

Comment (marcosgdf) [reply htdocs/core/class/html.formmail.class.php]: Well, I explained it in the bug report. `FormMail::getEmailTemplate` is returning an array, as opposed to the rest of FormMail functions which return a ModelMail object. Therefore, I changed getEmailTemplate and the places were it was treated as an array in `html.formmail.class.php`. Please check lines from 995 to 1047

**Comment by frederic34** [comment]
2018-03-11 10_23_30-commande - firefox developer edition

**Comment by marcosgdf** [comment]
@frederic34 It should always return an object. Is it still happening?

**Comment by frederic34** [comment]
https://github.com/Dolibarr/dolibarr/commit/df185de6aea7f485a3e641652ce4cc1c5597a98c

**Pull request merged**